kubernetes/docs/includes/deploy-jars-k8s.rst

```diff
@@ -0,0 +1,36 @@
+Deploying Custom Kafka Connect JARs with Operator 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Kafka Connect <https://docs.confluent.io/current/connect/index.html>`__ employs an extensible programming model to faciliate `Data Conversions <https://docs.confluent.io/current/connect/concepts.html#converters>`__, `Single Message Transformations (SMTs) <https://docs.confluent.io/current/connect/concepts.html#transforms>`__, or the development of custom connectors if a `supported connector <https://www.confluent.io/hub/>`__ is not available.  The Connect `packaging model <https://docs.confluent.io/current/connect/devguide.html#packaging>`__ will require that you deploy an archive containing your custom JAR to the `Kafka Connect's plugin path <https://docs.confluent.io/current/connect/userguide.html#connect-installing-plugins>`__.
+
+This demonstration utilizes a `custom Connector to generate mock events <https://github.com/confluentinc/kafka-connect-datagen>`__ which is **not** installed by default in the Confluent Kafka Connect Docker images.  In order to utilize this plugin, we build a custom Docker image which derives from the base Confluent Operator based Docker image and install our the custom connector using the ``confluent-hub`` client.
+
+Here is the ``Dockerfile`` snippet used to build these images (`source <https://github.com/confluentinc/kafka-connect-datagen/blob/0.1.x/Dockerfile-operator-local>`__):
```

Comment:
To maintain a single source of truth, consider removing the snippet and just link to the Dockerfile itself.  That way future changes to the Dockerfile won't put this doc out of sync.

Reply:
it would be great if i could link to a file in GitHub and have it rendered in RST, but I'm not aware of a solution.  I like having the Dockefile snippet here so a user can pull out the example right here and use in their own use.  IMO this is a low risk thing that drifts, and if it does it seems minimal risk vs the benefit to the reader to see a working example right away.

Reply:
Sometimes that's true, and I'd say that's almost true in this case except this Dockerfile in particular is a bit overly customized -- no one is going to be able to copy/paste it as is because:

1. It uses env var like `${KAFKA_CONNECT_DATAGEN_VERSION}`
2. It assumes locally compiled datagen connector with the `COPY` command
3. As a result of #2, the confluent-hub command is from local instead of Confluent Hub

We actually don't want people to use that model.  We want people to use the model where it's just:

```
RUN confluent-hub install --no-prompt kafka-connect-datagen:0.1.6
```

Therefore, it may in fact do more harm than good to include the whole Dockerfile.

Reply:
IMO The Dockerfile should be parsable by a novice Dockerfile builder.  Variables are common in Dockefiles.  This whole section is about _customzied_ jars/connectors for SMT, transformations, etc... in those cases confluent hub will not help them.  Linking them to the same solution doesn't help them if they can't read it there either.

I suggest we either remove the example and don't link them or spend the time making a generic one, which may exist in the confluent docs somewhere already.

Reply:
We could just link out to the confluent-hub reference document, and remove the example entirely but link to the kafka-conenct-datagen repo generally instead of the specific Dockerfile if we don't think it would be helpful (I think the snippet is helpful).

https://docs.confluent.io/current/connect/managing/confluent-hub/command-reference/index.html